lib/vagrant-service-manager/command.rb

```
@@ -266,21 +266,21 @@ def print_vagrant_box_version(script_readable = false)
       end
 
       def restart_service(service)
-
-        if SCCLI_SERVICES.include? service
-          # TODO : Handle the case where user wants to pass extra arguments to OpenShift service
-          command = "sudo sccli #{service}"
-        else
-          command = "sudo systemctl restart #{service}"
-        end
-
-        with_target_vms(nil, { single_target: true}) do |machine|
-          machine.communicate.execute(command) do |type, error|
-            if type == :stderr
-              @env.ui.error(error)
-              exit 126
-            end
+        command = if SCCLI_SERVICES.include? service
+                    # TODO : Handle the case where user wants to pass extra
+                    # arguments to OpenShift service
+                    "sccli #{service}"
+                  else
+                    "systemctl restart #{service}"
+                  end
+
+        with_target_vms(nil, single_target: true) do |machine|
+          exit_code = machine.communicate.sudo(command)
+          unless exit_code.zero?
+            @env.ui.error "#{command} failed"
```

@budhrg That would not solve the issue I raised. The question is what we expect the user to do when starting/restarting of OpenShift fails through VSM? IMO if VSM is running the commands and we are getting errors , it should propagate the errors to users